cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -37,13 +39,41 @@
 namespace arrow {
 namespace compute {
 
-// Remove this when multiple keys and/or types is supported
-typedef int32_t KeyType;
+template <typename T, typename V = typename T::value_type>
```

Comment:
Why do we need to wrap std_find here?

Reply:
It's just a convenience for writing more compact code when the begin/end iterators to `std::find` are those of the container.